benchmark/fs/bench-readdir.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+const fs = require('fs');
+const print = process._rawDebug;
+
+const ITER = 1e3;
+var t = process.hrtime();
+
+(function r(n) {
+  if (--n <= 0)
+    return completed();
+  fs.readdir(__dirname + '/../../lib/', function() {
+    r(n);
+  });
+}(ITER));
+
+function completed() {
+  t = process.hrtime(t);
+  print(((t[0] * 1e6 + t[1] / 1e3) / ITER).toFixed(2) + ' us/op');
+}
```

Comment:
I'm curious about this test.  Did you check what the user/kernel CPU time breakdown is and how much it's idle?  I could see this test being part GC stress test too, with the string and closure it creates.

The callback probably goes through an adapter frame as well?  Not 100% sure if that happens for no argument functions.